src/aap_eda/tasks/project.py

```diff
@@ -31,11 +32,57 @@ def import_project(project_id: int):
     logger.info(f"Task complete: Import project ( project_id={project.id} )")
 
 
-@job("default")
+@job(PROJECT_TASKS_QUEUE)
 def sync_project(project_id: int):
     logger.info(f"Task started: Sync project ( {project_id=} )")
 
     project = models.Project.objects.get(pk=project_id)
     ProjectImportService().sync_project(project)
 
     logger.info(f"Task complete: Sync project ( project_id={project.id} )")
+
+
+# Started by the scheduler, unique concurrent execution on default queue
+def monitor_project_tasks():
+    job_id = "monitor_project_tasks"
+    unique_enqueue(PROJECT_TASKS_QUEUE, job_id, _monitor_project_tasks)
+
+
+def _monitor_project_tasks() -> None:
+    """Handle project tasks that are stuck.
+
+    Check if there are projects in PENDING state that doesn't have
+    any related job in the queue. If there are any, put them back
+    to the queue.
+    """
+    logger.info("Task started: Monitor project tasks")
+
+    queue = get_queue(PROJECT_TASKS_QUEUE)
+
+    # Filter projects that doesn't have any related job
+    pending_projects = models.Project.objects.filter(
+        import_state=models.Project.ImportState.PENDING
```

Comment:
Very valid question @bzwei: I thought in that possibility as well but I finally decided that the risk to have stuck projects in running state with the monitor task would be equal or greater than the running projects themselves. This would be the same question as: What happen if the server is restarted during the execution of the monitor task?

Here is my reasoning:
Once the sync/import task has started it should be relatively short.
Handled restarts:
- If redis is restarted, the import task (the worker) would be completed anyway 
- If the worker is restarted, it it very unlikely that the task would not end in time before reach the graceful timeout

Unhandled restarts:
- If redis is restarted, the import task (the worker) would be completed anyway
- If worker dies or it is restarted, eventually it would take the job again from the queue. 

The current approach takes the broker as the source of truth. Which would not cover the 100% percent of the cases, but mostly. It is not a matter to add the "running" state in the query, to be really protected against that case, we should inspect all the default workers to see if indeed there is a task running. I feel this is very unlikely, but we could implement it in later iterations if we have reports. I not sure if it's worth the level of complexity it requires with the benefit.

Reply:
Another approach would be re-enqueue the task regardless if it is really running or not. (the worse case would be running it twice)